### PR TITLE
Cw 602 nano bad rep

### PR DESCRIPTION
**Pull request by fossephate**
Issue Number (if Applicable): Fixes #

# Description

Please include a summary of the changes and which issue is fixed / feature is added. 

# Pull Request - Checklist  

- [ ] Initial Manual Tests Passed
- [ ] Double check modified code and verify it with the feature/task requirements
- [ ] Format code
- [ ] Look for code duplication
- [ ] Clear naming for variables and methods

### CODE REVIEW

File: lib/src/screens/nodes/widgets/node_form.dart [hunk 1]
```diff
@@ -91,6 +96,18 @@ class NodeForm extends StatelessWidget {
             ],
           ),
           SizedBox(height: 10.0),
+          Row(
+            children: <Widget>[
+              Expanded(
+                child: BaseTextFormField(
+                  controller: _pathController,
+                  hintText: "/path",
+                  validator: NodePathValidator(),
```
Comment (OmarHatem28) [inline lib/src/screens/nodes/widgets/node_form.dart:105]: wouldn't this make it required?
while it can be empty

**Review by OmarHatem28 — CHANGES_REQUESTED** [review]

File: cw_nano/lib/nano_client.dart [hunk 2]
```diff
@@ -434,4 +437,37 @@ class NanoClient {
...
+      body: jsonEncode({
+        "action": "rep_info",
+        "account": rep,
+      }),
+    );
+    try {
+      final N2Node node = N2Node.fromJson(json.decode(response.body) as Map<String, dynamic>);
+      return node.score ?? 100;
+    } catch (error) {
+      return 100;
```
[the remark this answers is not in the document]
Comment (fossephate) [reply cw_nano/lib/nano_client.dart:470]: defaulting to a score of 100 ensures that we don't accidentally show the rep warning if the api goes down or something goes wrong
We want to warn users to change their rep when possible, but it isn't so urgent that we should risk a false positive

File: assets/nano_node_list.yml
[the remark this answers is not in the document]
Comment (fossephate) [reply assets/nano_node_list.yml:1]: I got them from this list:
https://publicnodes.somenano.com
they're just community nodes so the uptime on them isn't guaranteed or anything but I just wanted to give users more options

File: lib/src/screens/nano/nano_change_rep_page.dart [hunk 3]
```diff
@@ -5,10 +5,12 @@ import 'package:cake_wallet/src/widgets/alert_with_one_action.dart';
 import 'package:cake_wallet/src/widgets/alert_with_two_actions.dart';
 import 'package:cake_wallet/store/settings_store.dart';
 import 'package:cake_wallet/themes/extensions/address_theme.dart';
+import 'package:cake_wallet/themes/extensions/cake_text_theme.dart';
 import 'package:cake_wallet/utils/payment_request.dart';
 import 'package:cake_wallet/utils/show_pop_up.dart';
 import 'package:cw_core/crypto_currency.dart';
 import 'package:cw_core/wallet_base.dart';
+import 'package:cw_nano/n2_node.dart';
```
[the remark this answers is not in the document]
Comment (fossephate) [reply lib/src/screens/nano/nano_change_rep_page.dart:13]: should I just move the class? if so, where?
it's not a function like in past cases so I'm not sure how it'd fit into the usual solution of putting a function in cw_nano.dart

**Review by OmarHatem28 — APPROVED** [review]

**Pull request merged**